docker/include/bin/start_faux

```diff
@@ -87,7 +87,23 @@ if [ -n "${options[xdhcp]}" ]; then
     fi
     ip addr show $intf_name
 
-    (while true; do ping -c 1 10.20.0.1; sleep 5; done) &
+    if [ -n "${options[opendns]}" ]; then
+        route add -net 172.17.0.0/16 $intf_name  # default docker ip range.
+        dhcp_dns=8.8.8.8
+        echo nameserver ${dhcp_dns#DNS=} > /etc/resolv.conf
+        # There is no deterministic way to find out the default gateway ip where running more than 1 faux device.
+        (
+            while true; do
+                if [ `route -n | grep UG | wc -l` -eq '1' ]; then 
+                    timeout 1 ping -c 1 `route -n | grep UG ` && continue
```

Comment:
why is timeout necessary?  ping should timeout automatically?

Reply:
default is 4 seconds, so it takes too long to go through the list